File: example/play/provider/app/service/UserInternalService.scala
package service

import poppet.example.model.User
import poppet.example.service.UserService
import scala.concurrent.Future

class UserInternalService extends UserService {
    override def findById(id: String): Future[User] = {
        //emulation of business logic
        Future.successful(User(id, "Antony"))
    }
}

File: example/spring/provider/src/poppet/example/spring/provider/service/UserInternalService.java
package poppet.example.spring.provider.service;

import org.springframework.stereotype.Service;
import poppet.example.spring.model.User;
import poppet.example.spring.service.UserService;

@Service
public class UserInternalService implements UserService {
    @Override
    public User findById(String id) {
        //emulation of business logic
        if ("1".equals(id)) return new User(id, "Antony");
        else throw new IllegalArgumentException("User is not found");
    }
}
